src/table/build_industry.h

```diff
@@ -71,17 +63,16 @@ static const IndustryTileTable _tile_table_coal_mine_3[] = {
 	MK(2, 0, 6),
 	MK(2, 1, 4),
 	MK(2, 2, 3),
-	MKEND
 };
 
-static const IndustryTileTable * const _tile_table_coal_mine[] = {
+static const std::vector<IndustryTileLayout> _tile_table_coal_mine {
```

Comment:
would it not be better to use std::array for these?

Actually, wonder how hard it would be to use std::array for the statically defined industry layouts...

Reply:
I don't think there's any advantage to using std::array since all the tile tables are different lengths. Either most of the arrays would be too large, or they'd have to be converted to std::vector later anyway.
The memory savings from using std::vector should be larger than the performance gains from using std::array, since this isn't code typically run in a tight loop.